### ISSUE TITLE
LOAD doesn't store absolute pathnames for source locations

### ISSUE THREAD

**Issue by stassats** [drmeister/clasp#113]
`(list (load "/tmp/foo.lisp") (ext:compiled-function-file #'foo) (load (compile-file "/tmp/foo.lisp")) (ext:compiled-function-file #'foo))`
=> (T "foo.lisp" T "/tmp/foo.lisp")

**Comment by Shinmera** [comment]
Cannot confirm on b05884f due to a different issue:
```
> (load #p"~/a.lisp")

T
> (foo)

"hi" 
"hi"
> (ext:compiled-function-file #'foo)

"/home/linus/a.lisp"
1
> (load (compile-file #p"~/a.lisp"))
Writing fasl file to: #P"/home/linus/a.fasl"
Linking #P"/home/linus/a.bc"
Linking app-resources:lib;release;intrinsics_bitcode_mps.sbc
Generating object file  /home/linus/a.lbc  -->  /home/linus/a.o  reloc-model: RELOC-MODEL-PIC-
T
> (ext:compiled-function-file #'foo)
The MPS detected a problem!
src/mps/code/lockli.c:139: MPS ASSERTION FAILED: res == 0
See the "Assertions" section in the reference manual:
http://www.ravenbrook.com/project/mps/master/manual/html/topic/error.html#assertions
The MPS detected a problem!
src/mps/code/lockli.c:143: MPS ASSERTION FAILED: lock->claims == 0
See the "Assertions" section in the reference manual:
http://www.ravenbrook.com/project/mps/master/manual/html/topic/error.html#assertions
The MPS detected a problem!
src/mps/code/shield.c:224: MPS ASSERTION FAILED: !arena->insideShield
See the "Assertions" section in the reference manual:
http://www.ravenbrook.com/project/mps/master/manual/html/topic/error.html#assertions
The MPS detected a problem!
src/mps/code/shield.c:226: MPS ASSERTION FAILED: !arena->suspended
See the "Assertions" section in the reference manual:
http://www.ravenbrook.com/project/mps/master/manual/html/topic/error.html#assertions
[1]    27904 segmentation fault (core dumped)  ./clasp_mps_o
```
a.lisp contains:
```
(defun foo ()
  (print "hi"))
```

**Comment by drmeister** [comment]
The MPS problems are MPS/Clasp problems - nothing to do with the pathnames.  I fixed the absolute pathnames for functions that are LOAD'ed in df5b26ca3186b1d7adc1ee248d8e057a8205f26f

**Comment by Shinmera** [comment]
Fix confirmed on 9c24f9a45d0bd4bd4a80cdcac18b7cee012947fb
```
> (load #p"~/a.lisp")

T
> (ext:compiled-function-file #'foo)

#P"/home/linus/a.lisp"
1
> (load (compile-file #p"~/a.lisp"))
Writing fasl file to: #P"/home/linus/a.fasl"
Linking #P"/home/linus/a.bc"
Linking app-contents:execs;boehm;release;bin;intrinsics_bitcode.sbc
warning: Linking two modules of different target triples: /home/linus/.cache/shirakumo/autobuild/clasp-testing/.autobuild/9c24f9a45d0bd4bd4a80cdcac18b7cee012947fb/build/clasp/Contents/execs/boehm/release/bin/intrinsics_bitcode.sbc' is 'x86_64-unknown-linux-gnu' whereas 'a' is 'x86_64-pc-linux-gnu'

Generating object file  /home/linus/a.lbc  -->  /home/linus/a.o  reloc-model: RELOC-MODEL-PIC-
T
> (ext:compiled-function-file #'foo)

#P"/home/linus/a.lisp"
1
```
